lib/guardian.rb

```diff
@@ -391,9 +391,11 @@ def can_export_entity?(entity)
   end
 
   def can_ignore_user?(user_id)
-    @user.id != user_id &&
-      (@user.staff? || @user.trust_level >= TrustLevel.levels[:member]) &&
-      User.where(id: user_id, admin: false, moderator: false).exists?
+    can_ignore_users? && @user.id != user_id && User.where(id: user_id, admin: false, moderator: false).exists?
+  end
+
+  def can_ignore_users?
+    SiteSetting.ignore_user_enabled? && (@user.staff? || @user.trust_level >= TrustLevel.levels[:member])
```

Comment:
I believe it's valid to create a `Guardian` with a `nil` user, for anonymous users. It looks like you'd get a `nil` error here.

Reply:
If a nil user is passed, there is this line:

```ruby
    @user = user.presence || AnonymousUser.new
```

When I do: `::Guardian::AnonymousUser.new.staff?`, it returns false, but it fails for `trust_level`

Reply:
Perhaps I can exit early: `return false if anonymous?`

Reply:
So you could fix it either by adding `0` to the Anonymous user's trust level, or by doing what we usually do which is `return false if @user.blank?`

Reply:
Won't it be better as: `return false if anonymous?`

Reply:
Sure, I forgot about that helper method. even better :)